### PR TITLE
[release-ocm-2.9] MGMT-18313: Replace golang base image as it is based on Centos Linux 7

### PR DESCRIPTION
**Pull request by danmanor**
This PR moves centos base image from stream8 to stream 9 as well as stream8 is EOL

### CODE REVIEW

File: Dockerfile.assisted-installer [hunk 1]
```diff
@@ -1,4 +1,8 @@
-FROM registry.ci.openshift.org/openshift/release:golang-1.19 AS builder
+FROM registry.access.redhat.com/ubi9/go-toolset:1.18 AS builder
```
Comment (eifrach) [inline Dockerfile.assisted-installer:1]: why are you downgrading the Go version here?
is it by mistake ?

**Review by eifrach — CHANGES_REQUESTED** [review]

File: Dockerfile.assisted-installer (hunk 1 above)
Comment (danmanor) [reply Dockerfile.assisted-installer:1]: This is the actual go version, look at `go.mod`

**Pull request merged**